ConsoleApplication task Lelek

itevas opened the pull request:
Console App task Lelek

Commit:
Console App task

inventorsoft-academy commented on pom.xml:
```diff
@@ -15,4 +15,54 @@
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-mail</artifactId>
+            <version>2.1.9.RELEASE</version>
```
don't hardcode version. use implicit version that will be provided by spring boot parent

inventorsoft-academy commented on src/main/java/com.lelek.springBoot/ApplicationConfiguration.java:
```diff
@@ -0,0 +1,37 @@
+
+@Slf4j
+@Configuration
+@PropertySource("classpath:mail.properties")
```
consider moving mail properties in `application.properties`. then you will get them loaded for free. try to rely on standard behaviour that framework provides you

inventorsoft-academy commented on src/main/java/com.lelek.springBoot/ApplicationConfiguration.java:
```diff
@@ -0,0 +1,37 @@
+    private Environment environment;
+
+    @Bean
+    public JavaMailSender javaMailSender() {
```
This bean is preconfigured for you by `spring-boot-starter-mail`. all you need to do is adding relevant configuration into `application.properies`

inventorsoft-academy commented on src/main/java/com.lelek.springBoot/Sender.java:
```diff
@@ -0,0 +1,67 @@
+    @Autowired
+    private JavaMailSender javaMailSender;
+
+    private List<MySimpleMailMessage> getAllMessages() throws IOException {
```
I feel like this class is doing more than it should. As the name states, this component is responsible for sending the messages. Storage is a different thing. As we discussed, we want to have an ability to switch storage implementation in future: now it's file-based, but we want to support database. If you will stick to this approach this component will quickly become overloaded by functionality and hard to work with

inventorsoft-academy commented on src/main/java/com.lelek.springBoot/MySimpleMailMessage.java:
```diff
@@ -0,0 +1,54 @@
+@Slf4j
+@Getter
+@Setter
+@Component
```
If you're using `@Component` annotation without specifying scope, default one will be singleton. But this class has state which won't play well in multithreaded environment: imaging multiple threads will try to update sent property; what will be the outcome? In general, don't make stateful classes as spring components. This can still be created via new operator on demand. Consolidate persistence/storage logic inside one component: save from this class and read all from Sender should be in one place, so you know where to look if you want to understand how messages are stored

Commit:
added spring boot parent to POM
moved mail.properties to application.properties
removed config class

Commit:
added spring boot parent to POM
moved mail.properties to application.properties
removed config class
added dao, dto, model and service packages

Commit:
WebApplication task fixed some issues

Commit:
ConsoleApplication task fixed some issues
